lib/packages/docutils/rst.nim

```diff
@@ -514,10 +525,17 @@ proc defaultFindFile*(filename: string): string =
   if fileExists(filename): result = filename
   else: result = ""
 
+proc defaultRoleKind(options: RstParseOptions): RstNodeKind =
+  if roNimFile in options: rnInlineCode else: rnInlineLiteral
+proc defaultRole(options: RstParseOptions): string =
+  if roNimFile in options: "nim" else: "literal"
```

Comment:
add
```nim
type BuiltinRoles = enum
  brUnknown = "unknown"
  brNim = "nim"
  brLiteral = "literal"
  brCode = "code"
```

and then use it everytime you're using one of those builtin roles, it's more typesafe eg:
```nim
if roNimFile in options: $brNim else: $brLiteral
```

Reply:
too little profit for a new type. Most roles are met only once and immediately converted to RstNodeKind. The only exception is "nim" and "literal", which occur twice, but both occurrences are within a few lines.